### PR TITLE
[#IOPID-1221] Enable immutability policy on `spid_logs_storage` and `lv_audit_logs_storage`

### PR DESCRIPTION
**Pull request by gquadrati**
### List of changes

- Update version
- Enable immutability policy

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: src/domains/ioweb-common/03_storage.tf [hunk 1]
```diff
@@ -31,6 +44,26 @@ module "spid_logs_storage_customer_managed_key" {
   storage_principal_id = module.spid_logs_storage.identity.0.principal_id
 }
 
+resource "azurerm_storage_management_policy" "spid_logs_storage_management_policy" {
+  storage_account_id = module.spid_logs_storage.id
+
+  rule {
+    name    = "deleteafterdays"
```
Comment (BurnedMarshal) [inline src/domains/ioweb-common/03_storage.tf:51]:
```suggestion
    name    = "deleteafter2yrs"
```

File: src/domains/ioweb-common/03_storage.tf [hunk 2]
```diff
@@ -31,6 +44,26 @@ module "spid_logs_storage_customer_managed_key" {
...
+    enabled = true
+    filters {
+      prefix_match = [
+        "spidlogs",
+      ]
+      blob_types = ["blockBlob"]
+    }
+    actions {
+      base_blob {
```
Comment (BurnedMarshal) [inline src/domains/ioweb-common/03_storage.tf:60]: Apply the policy even for shapshot and version

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]
A couple of issues may affect this apply. The module cannot handle properly the replace of the resource. See the known issues into the [module readme](https://github.com/pagopa/terraform-azurerm-v3/tree/main/storage_account#known-issues).

File: src/domains/citizen-auth-common/03_storage.tf [hunk 3]
```diff
@@ -142,6 +146,112 @@ resource "azurerm_storage_container" "lv_audit_logs_storage_logs" {
...
 
+
+
+###
+# Immutable LV Audit Log Storage
+###
+module "immutable_lv_audit_logs_storage" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3//storage_account?ref=v7.32.1"
+
+  name                          = replace(format("%s-imm-lv-logs-st", local.product), "-", "")
```
Comment (BurnedMarshal) [inline src/domains/citizen-auth-common/03_storage.tf:157]: Maybe for better readability we should put `immutable` tag at the end of the storage name
es. `"%s-lv-logs-imm-st"`
So we have `[project]-[context]-[options]-[type]`

File: src/domains/ioweb-common/03_storage.tf [hunk 4]
```diff
@@ -60,4 +63,107 @@ resource "azurerm_storage_container" "spid_logs" {
...
+}
+
+
+######################
+# Immutable SPID LOGS Storage
+######################
+module "immutable_spid_logs_storage" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3//storage_account?ref=v7.32.1"
+
+  name                          = replace(format("%s-imm-spid-logs-st", local.project), "-", "")
```
Comment (BurnedMarshal) [inline src/domains/ioweb-common/03_storage.tf:75]: ditto

File: src/domains/ioweb-common/10_spid_login.tf [hunk 5]
```diff
@@ -112,7 +112,7 @@ module "spid_login" {
     # Spid logs
     ENABLE_SPID_ACCESS_LOGS             = true
     SPID_LOGS_ENABLE_PAYLOAD_ENCRYPTION = false
-    SPID_LOGS_STORAGE_CONNECTION_STRING = module.spid_logs_storage.primary_connection_string
+    SPID_LOGS_STORAGE_CONNECTION_STRING = module.immutable_spid_logs_storage.primary_connection_string
```
Comment (BurnedMarshal) [inline src/domains/ioweb-common/10_spid_login.tf:115]: What if we change the pointing to the resources on another PR?

**Review by BurnedMarshal — APPROVED** [review]
LGTM, only some optional comment

File: src/domains/ioweb-common/10_spid_login.tf (hunk 5 above)
Comment (gquadrati) [reply src/domains/ioweb-common/10_spid_login.tf]: Done in [fix based on comments](https://github.com/pagopa/io-infra/pull/778/commits/1e4268778c5a0b35cfe764917907553e84ffb6de)

File: src/domains/ioweb-common/03_storage.tf (hunk 4 above)
Comment (gquadrati) [reply src/domains/ioweb-common/03_storage.tf]: Done in [fix based on comments](https://github.com/pagopa/io-infra/pull/778/commits/1e4268778c5a0b35cfe764917907553e84ffb6de)

File: src/domains/citizen-auth-common/03_storage.tf (hunk 3 above)
Comment (gquadrati) [reply src/domains/citizen-auth-common/03_storage.tf]: Done in [fix based on comments](https://github.com/pagopa/io-infra/pull/778/commits/1e4268778c5a0b35cfe764917907553e84ffb6de)

**Review by BurnedMarshal — APPROVED** [review]

**Pull request merged**